xbmc/platform/win32/filesystem/Win32File.cpp

```diff
@@ -466,7 +466,15 @@ int CWin32File::Stat(const CURL& url, struct __stat64* statData)
     return -1;
   }
 
-  if (pathnameW.length() <= 6) // 6 is length of "\\?\x:"
+  size_t pathLength{pathnameW.length()};
+  // stat of directories requires the removal of the trailing slash
+  if (pathnameW[pathLength - 1] == L'\\')
+  {
+    pathnameW.erase(pathLength - 1);
+    --pathLength;
```

Comment:
`--pathLength` it's a bit ugly, maybe remove

and use `pathnameW.length()` in L477 and then `pathLength` declaration can be `const size_t`

Reply:
Or even better:

```c++
  const size_t last{pathnameW.length() - 1};
  // stat of directories requires the removal of the trailing slash
  if (pathnameW[last] == L'\\')
    pathnameW.erase(last);
```

Reply:
That's what I originally had, and then noticed there was another use of the length in the function. (L477).
Seemed unnecessary to throw away the length and calculate it again. But I don't see a prettier way.

There is this, prettier but maybe less efficient:
```c++
  // stat of directories requires the removal of the trailing slash
  if (pathnameW.back() == L'\\')
    pathnameW.pop_back();

  if (pathnameW.length() <= 6) // 6 is length of "\\?\x:"
```